src/index.js

```diff
@@ -2,0 +1,38 @@
+// 'use strict';
+import express from 'express';
+import cookieParser from 'cookie-parser';
+import cors from 'cors';
+import { authRouter } from './routes/auth.route.js';
+import { userRoute } from './routes/user.route.js';
+import { resetRoute } from './routes/reset.route.js';
+import { errorMiddleware } from './middleware/error.middleware.js';
+import { profileRoute } from './routes/profile.route.js';
+
+const app = express();
+const PORT = process.env.PORT || 3005;
+
+app.use(express.json());
+app.use(cookieParser());
+
+app.use(
+  cors({
+    origin: process.env.CLIENT_HOST || 'http://localhost:3000',
+    credentials: true,
+  }),
+);
+
+app.use(authRouter);
+app.use(resetRoute);
+app.use(profileRoute);
+app.use('/users', userRoute);
```

Comment:
Consider using a consistent base path for all routes. For example, you could use '/auth' for authRouter, '/reset' for resetRoute, and '/profile' for profileRoute, similar to how '/users' is used for userRoute. This would make the API more intuitive and organized.